packages/discord.js/src/structures/BaseChannel.js

```diff
@@ -155,6 +155,14 @@ class BaseChannel extends Base {
     return 'availableTags' in this;
   }
 
+  /**
+   * Indicates whether this channel is {@link TextBasedChannels text-based} and can have messages sent in it.
+   * @returns {boolean}
+   */
+  isSendable() {
+    return this.isTextBased() && 'send' in this;
+  }
```

Comment:
Is the first check here needed? If a channel had a send method it should be considered sendable. Even if for whatever reason it‘s not TextBased (although no channel type that does that currently exists).

Reply:
I'd rather keep it in, if in the future for god-knows what reason we get a channel with a .send that doesn't do messages XD

Reply:
Then you can simply chain both typeguards as a user… but I‘m good with either way

Reply:
Leaning towards @Qjuh, but I am majorly impartial here.